server/src/lib/server.rs

```diff
@@ -168,6 +177,106 @@ fn slot_to_bdf(slot: api::Slot, ty: SlotType) -> Result<pci::Bdf> {
     }
 }
 
+fn create_in_memory_cloud_init_iso(
+    log: &Logger,
+    cloud_init: &api::CloudInit,
+) -> anyhow::Result<Vec<u8>> {
+    info!(log, "creating temp dir");
+
+    let temp_dir = tempdir()?;
```

Comment:
Probably not as much of an issue as each propolis instance will run in its own zone anyways, but it'd be nice if we could do this entirely in RAM without having to create things on disk first. Worry here being a malicious actor swapping out the file between when we write and when we read it.

Anyways, don't take this is a blocker--I imagine it'd require a lot more work to re architecture the iso crate.

Reply:
Changing the mkisofs-rs crate to operate on `Read + Write + Seek` instead of `File` for the output file is straight forward but it's not immediately clear to me how to nicely do it for input files. The Opt struct of the crate takes input directories to walk for files instead of files themselves.

I was thinking of adding a "input_files_raw" option which would be a Vec of `Read + Seek` but I'm not sure I like that way.